### PR TITLE
API Remove ComposerUpdate model and shift to a DataExtension on ComposerPackageVersion

### LINKED ISSUE #47: Separate constraint and hash from the "Version" field of ComposerPackageVersion
Requires #46 first

If using `dev-master` as a constraint, the Version column also contains the hash that is installed.

Ideally this would be in a separate column, e.g. InstalledVersion and InstalledHash for example

image

### PR DESCRIPTION
**Pull request by robbieaverill**
~~**Contains the changes in #21 as well - when merged this may need a rebase**~~

This changes:

* Removes the ComposerUpdate model and shifts the logic into a DataExtension which is applied to ComposerPackageVersion (DataObject), adding the extra columns to that model instead (resolves #https://github.com/bringyourownideas/silverstripe-composer-update-checker/issues/25)
* Separates the installed version and the installed hash into separate columns (resolves https://github.com/bringyourownideas/silverstripe-maintenance/issues/47)

Requires https://github.com/bringyourownideas/silverstripe-maintenance/pull/51

### CODE REVIEW

**Comment by robbieaverill** [comment]
~~This PR will need to be updated once https://github.com/bringyourownideas/silverstripe-maintenance/pull/49 is merged~~ done

**Comment by robbieaverill** [comment]
Requires https://github.com/bringyourownideas/silverstripe-maintenance/pull/51

File: src/Tasks/CheckComposerUpdatesTask.php [hunk 1]
```diff
@@ -340,25 +340,24 @@ protected function isStableEnough($currentStability, $possibleStability)
...
-            $update = ComposerUpdate::create();
+            $update = Package::create();
             $update->Name = $package;
-        }
-
-        // If installed is dev-master get the hash
-        if ($installed === 'dev-master') {
-            $localPackage = $this->getLocalPackage($package);
-            $installed = $localPackage->source->reference;
+            $update->Version = $installed;
```
Comment (NightJar) [inline src/Tasks/CheckComposerUpdatesTask.php:355]: Will also need to set `Description` and `Type` if this is to replace the default refresh task included with `silverstripe-maintenance`

**Review by NightJar — COMMENTED** [review]
Looks fine, but like it might be a part of a greater piece of work. If this is the case then my comment isn't of much consequence, but if not then it should be addressed.

File: src/Tasks/CheckComposerUpdatesTask.php (hunk 1 above)
Comment (robbieaverill) [reply src/Tasks/CheckComposerUpdatesTask.php:355]: It’s not aimed at that, expect more follow ups to come

Comment (NightJar) [reply src/Tasks/CheckComposerUpdatesTask.php:355]: In this PR, or just in general?
(ie. merge now?)

Comment (robbieaverill) [reply src/Tasks/CheckComposerUpdatesTask.php:355]: Further PRs (there have been a few building up)

Comment (NightJar) [reply src/Tasks/CheckComposerUpdatesTask.php:355]: Sweet I'll merge then :)

**Pull request merged**